### ISSUE TITLE
The row limit set for the chart was reached.

### ISSUE THREAD

**Issue by Habeeb556** [apache/superset#29409]
### Bug description

I got an error message chart limit when creating a chart from sqllab:

`The row limit set for the chart was reached. The chart may show partial data.`

### How to reproduce the bug

Create a chart from sqllab.

### Screenshots/recordings

![image](https://github.com/apache/superset/assets/35889071/b909cb12-5cca-48f7-9566-977a283f2600)

### Superset version

4.0.1

### Python version

3.11

### Node version

I don't know

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by Habeeb556** [comment]
@dosu

How can I adjust this limit? Which file path should I use?

**Comment by Habeeb556** [comment]
@dosu

It doesn't affect the chart; it impacts SQLLab directly.

**Comment by Habeeb556** [comment]
Not solved, still got chart error `Limit reached`.

**Comment by rusackas** [comment]
The default limit in SQL Lab is 1000... did you change it if needed?

<img width="1306" alt="Pasted_Image_7_1_24__9_46 AM" src="https://github.com/apache/superset/assets/812905/5c378bbc-0f90-4569-b9d3-8280a381d670">

**Comment by Habeeb556** [comment]
Yes, I know that, but the limitation is within the chart creation itself. I have already open limit in SQL Lab to 100k, but when creating a chart from result, I encountered the following message indicating that the 1k limit has been reached:

`The row limit set for the chart was reached. The chart may show partial data.`

![image](https://github.com/apache/superset/assets/35889071/52e1d7f6-b6f3-4bc9-9662-e081cb7e297d)

**Comment by rusackas** [comment]
Can you give us a reproducible test case using one of the example datasets? That'll help me (or anyone) more able to validate that it's happening this way on our instance(s) and see if we can find a solution/workaround.

**Comment by Habeeb556** [comment]
1- Run any query in SQLLAB fetch 10K rows.
2- Then click on `CREATE CHART` to explore the result.
3- You will get only 1K rows limit.

Additionally, I am using this method to download the result as an XLSX file from the chart. Therefore, I request a discussion #29411 to add a "Download to XLSX" feature, similar to the "Download to CSV" button.

**Comment by Habeeb556** [comment]
Any suggestion here....

**Comment by mistwan** [comment]
you can alter there ， 
> superset/common/query_object.py:150
```python
# self.row_limit = row_limit
from superset import app
self.row_limit = app.config["ROW_LIMIT"]
```

then try superset `custome config`  > `page length` 

![微信图片_20241213155945](https://github.com/user-attachments/assets/f3c0f375-8d04-4dd4-bc5a-b7a78ce2cacb)

**Comment by Habeeb556** [comment]
Wow! it worked @mistwan thaanks.

![image](https://github.com/user-attachments/assets/1809792c-06f2-4989-ba0a-7bdae7d234d5)

**Comment by Habeeb556** [comment]
@rusackas I think this should be reviewed as it might help mitigate the issue in the future.

**Comment by rusackas** [comment]
I've totally lost track of what exactly needs to be changed here (if anything, in current versions) - is there a PR to be had, or a docs change? This hit my inactive issues radar, but I'd love to have proper closure here if we can make things better.

**Comment by rusackas** [comment]
@dosu can you review this thread and suggest what fixes we should make in the codebase, or at least what docs we might want to add to the Superset website?

**Comment by Habeeb556** [comment]
Hi @rusackas, I believe this issue can be closed as it's not actually a bug. The behavior I raise is expected: when creating a chart, the default row limit is set to 1,000, which is reasonable for display purposes.

In my case, I needed to download the query results from SQL Lab in Excel format, so I temporarily increased the row limit beyond 1,000. This led to slower loading and caching, which is not recommended for regular use. I only applied this change for testing—not for publishing.
